sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/Mqtt/MqttConnectEventErrorResponseProperties.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Azure.WebPubSub.Common;
+
+#nullable enable
+
+/// <summary>
+/// Represents the properties of an MQTT connection failure response.
+/// </summary>
+[DataContract]
+public class MqttConnectEventErrorResponseProperties
+{
+    internal const string CodeProperty = "code";
+    internal const string ReasonProperty = "reason";
+    internal const string UserPropertiesProperty = "userProperties";
```

Comment:
shall we extend userProperties only and leave code/reason in upper class share with ErrorEventResponse?

Reply:
The code here is an MQTT specified code, whose valid values are defined in MQTT protocols, so can't be shared with the based class.

The `reason` is to be sent to the MQTT clients. MQTT upstream protocols define the `reason` field while the Web PubSub's upstream protocol doesn't. Therefore, we keep the `reason` field in MQTT fields only.